GO version 1.23 not available

**Describe the bug**
Currently there is no way to get GO 1.23 to work. 

**To reproduce**
Use the following devenv yaml / nix:
```yaml
# devenv.yaml
inputs:
  nixpkgs:
    url: github:cachix/devenv-nixpkgs/rolling
  nixpkgs-unstable:
    url: github:nixos/nixpkgs/nixpkgs-unstable
```

```nix
# devenv.nix
{ pkgs, inputs, ... }:
let
  pkgs-unstable = import inputs.nixpkgs-unstable { system = pkgs.stdenv.system; };
in
{
  languages.go.package = pkgs-unstable.go;
}
```

Then just `devenv shell`:
```
devenv shell -- go version
• Building shell ...
• Using Cachix: devenv
error:
       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:9:12:
            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'devenv-shell'
         whose name attribute is located at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'DEVENV_PROFILE' of derivation 'devenv-shell'

       … while calling the 'getAttr' builtin
         at <nix/derivation-internal.nix>:19:19:
           18|       value = commonAttrs // {
           19|         outPath = builtins.getAttr outputName strict;
             |                   ^
           20|         drvPath = strict.drvPath;

       … while calling the 'derivationStrict' builtin
         at <nix/derivation-internal.nix>:9:12:
            8|
            9|   strict = derivationStrict drvAttrs;
             |            ^
           10|

       … while evaluating derivation 'devenv-profile'
         whose name attribute is located at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/stdenv/generic/make-derivation.nix:331:7

       … while evaluating attribute 'passAsFile' of derivation 'devenv-profile'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/trivial-builders/default.nix:69:9:
           68|         inherit buildCommand name;
           69|         passAsFile = [ "buildCommand" ]
             |         ^
           70|           ++ (derivationArgs.passAsFile or [ ]);

       … while evaluating the attribute 'passAsFile'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:76:5:
           75|     # XXX: The size is somewhat arbitrary
           76|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |     ^
           77|   }

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:76:18:
           75|     # XXX: The size is somewhat arbitrary
           76|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |                  ^
           77|   }

       … in the argument of the not operator
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:76:48:
           75|     # XXX: The size is somewhat arbitrary
           76|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |                                                ^
           77|   }

       … while calling the 'lessThan' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:76:48:
           75|     # XXX: The size is somewhat arbitrary
           76|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |                                                ^
           77|   }

       … while calling the 'stringLength' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:76:21:
           75|     # XXX: The size is somewhat arbitrary
           76|     passAsFile = if builtins.stringLength pkgs >= 128*1024 then [ "pkgs" ] else [ ];
             |                     ^
           77|   }

       … while calling the 'toJSON' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:58:12:
           57|             nativeBuildInputs buildInputs;
           58|     pkgs = builtins.toJSON (map (drv: {
             |            ^
           59|       paths =

       … while calling the 'map' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/build-support/buildenv/default.nix:58:29:
           57|             nativeBuildInputs buildInputs;
           58|     pkgs = builtins.toJSON (map (drv: {
             |                             ^
           59|       paths =

       … from call site
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/top-level.nix:15:13:
           14|     name = "devenv-profile";
           15|     paths = lib.flatten (builtins.map drvOrPackageToPaths config.packages);
             |             ^
           16|     ignoreCollisions = true;

       … while calling 'flatten'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:383:13:
          382|   */
          383|   flatten = x:
             |             ^
          384|     if isList x

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:384:5:
          383|   flatten = x:
          384|     if isList x
             |     ^
          385|     then concatMap (y: flatten y) x

       … while calling the 'isList' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:384:8:
          383|   flatten = x:
          384|     if isList x
             |        ^
          385|     then concatMap (y: flatten y) x

       … while calling the 'map' builtin
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/top-level.nix:15:26:
           14|     name = "devenv-profile";
           15|     paths = lib.flatten (builtins.map drvOrPackageToPaths config.packages);
             |                          ^
           16|     ignoreCollisions = true;

       … from call site
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/top-level.nix:15:59:
           14|     name = "devenv-profile";
           15|     paths = lib.flatten (builtins.map drvOrPackageToPaths config.packages);
             |                                                           ^
           16|     ignoreCollisions = true;

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/attrsets.nix:1171:18:
         1170|         mapAttrs
         1171|           (name: value:
             |                  ^
         1172|             if isAttrs value && cond value

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/attrsets.nix:1174:18:
         1173|             then recurse (path ++ [ name ]) value
         1174|             else f (path ++ [ name ]) value);
             |                  ^
         1175|     in

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:242:72:
          241|           # For definitions that have an associated option
          242|           declaredConfig = mapAttrsRecursiveCond (v: ! isOption v) (_: v: v.value) options;
             |                                                                        ^
          243|

       … while evaluating the attribute 'value'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:809:9:
          808|     in warnDeprecation opt //
          809|       { value = builtins.addErrorContext "while evaluating the option `${showOption loc}':" value;
             |         ^
          810|         inherit (res.defsFinal') highestPrio;

       … while evaluating the option `packages':

       … while evaluating the attribute 'mergedValue'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:844:5:
          843|     # Type-check the remaining definitions, and merge them. Or throw if no definitions.
          844|     mergedValue =
             |     ^
          845|       if isDefined then

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:846:59:
          845|       if isDefined then
          846|         if all (def: type.check def.value) defsFinal then type.merge loc defsFinal
             |                                                           ^
          847|         else let allInvalid = filter (def: ! type.check def.value) defsFinal;

       … while calling 'merge'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:546:20:
          545|       check = isList;
          546|       merge = loc: defs:
             |                    ^
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:

       … while calling the 'map' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:547:9:
          546|       merge = loc: defs:
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
             |         ^
          548|           imap1 (m: def':

       … while calling the 'filter' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:547:27:
          546|       merge = loc: defs:
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
             |                           ^
          548|           imap1 (m: def':

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:547:35:
          546|       merge = loc: defs:
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
             |                                   ^
          548|           imap1 (m: def':

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:547:38:
          546|       merge = loc: defs:
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
             |                                      ^
          548|           imap1 (m: def':

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:334:29:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                             ^
          335|

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:334:32:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                ^
          335|

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:548:21:
          547|         map (x: x.value) (filter (x: x ? value) (concatLists (imap1 (n: def:
          548|           imap1 (m: def':
             |                     ^
          549|             (mergeDefinitions

       … while evaluating the attribute 'optionalValue'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:856:5:
          855|
          856|     optionalValue =
             |     ^
          857|       if isDefined then { value = mergedValue; }

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:857:7:
          856|     optionalValue =
          857|       if isDefined then { value = mergedValue; }
             |       ^
          858|       else {};

       … while evaluating the attribute 'values'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:838:9:
          837|       in {
          838|         values = defs''';
             |         ^
          839|         inherit (defs'') highestPrio;

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:834:11:
          833|           # Avoid sorting if we don't have to.
          834|           if any (def: def.value._type or "" == "order") defs''.values
             |           ^
          835|           then sortProperties defs''.values

       … while calling the 'any' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:834:14:
          833|           # Avoid sorting if we don't have to.
          834|           if any (def: def.value._type or "" == "order") defs''.values
             |              ^
          835|           then sortProperties defs''.values

       … while evaluating the attribute 'values'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:937:7:
          936|     in {
          937|       values = concatMap (def: if getPrio def == highestPrio then [(strip def)] else []) defs;
             |       ^
          938|       inherit highestPrio;

       … while calling the 'concatMap' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:937:16:
          936|     in {
          937|       values = concatMap (def: if getPrio def == highestPrio then [(strip def)] else []) defs;
             |                ^
          938|       inherit highestPrio;

       … while calling the 'concatMap' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:824:17:
          823|         # Process mkMerge and mkIf properties.
          824|         defs' = concatMap (m:
             |                 ^
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:824:28:
          823|         # Process mkMerge and mkIf properties.
          824|         defs' = concatMap (m:
             |                            ^
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))

       … while calling the 'map' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:825:11:
          824|         defs' = concatMap (m:
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |           ^
          826|         ) defs;

       … while evaluating definitions from `«github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/languages/go.nix':

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:825:137:
          824|         defs' = concatMap (m:
          825|           map (value: { inherit (m) file; inherit value; }) (builtins.addErrorContext "while evaluating definitions from `${m.file}':" (dischargeProperties m.value))
             |                                                                                                                                         ^
          826|         ) defs;

       … while calling 'dischargeProperties'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:896:25:
          895|   */
          896|   dischargeProperties = def:
             |                         ^
          897|     if def._type or "" == "merge" then

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/modules.nix:897:5:
          896|   dischargeProperties = def:
          897|     if def._type or "" == "merge" then
             |     ^
          898|       concatMap dischargeProperties def.contents

       … while evaluating the attribute 'value'
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/types.nix:552:38:
          551|               elemType
          552|               [{ inherit (def) file; value = def'; }]
             |                                      ^
          553|             ).optionalValue

       … while calling the 'elemAt' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/lists.nix:334:43:
          333|   */
          334|   imap1 = f: list: genList (n: f (n + 1) (elemAt list n)) (length list);
             |                                           ^
          335|

       … from call site
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/languages/go.nix:37:8:
           36|       # Required by vscode-go
           37|       (buildWithSpecificGo pkgs.delve)
             |        ^
           38|

       … while calling 'buildWithSpecificGo'
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/languages/go.nix:8:25:
            7|
            8|   buildWithSpecificGo = pkg: pkg.override {
             |                         ^
            9|     buildGoModule = pkgs."buildGo${goVersion}Module".override {

       … while calling a functor (an attribute set with a '__functor' attribute)
         at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/languages/go.nix:8:30:
            7|
            8|   buildWithSpecificGo = pkg: pkg.override {
             |                              ^
            9|     buildGoModule = pkgs."buildGo${goVersion}Module".override {

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/trivial.nix:895:7:
          894|     { # TODO: Should we add call-time "type" checking like built in?
          895|       __functor = self: f;
             |       ^
          896|       __functionArgs = args;

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:144:34:
          143|       # Re-call the function but with different arguments
          144|       overrideArgs = mirrorArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                  ^
          145|       # Change the result of the function call by applying g to it

       … while calling a functor (an attribute set with a '__functor' attribute)
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:144:43:
          143|       # Re-call the function but with different arguments
          144|       overrideArgs = mirrorArgs (newArgs: makeOverridable f (overrideWith newArgs));
             |                                           ^
          145|       # Change the result of the function call by applying g to it

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/trivial.nix:895:7:
          894|     { # TODO: Should we add call-time "type" checking like built in?
          895|       __functor = self: f;
             |       ^
          896|       __functionArgs = args;

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:136:17:
          135|     in
          136|     mirrorArgs (origArgs:
             |                 ^
          137|     let

       … while evaluating a branch condition
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:148:7:
          147|     in
          148|       if isAttrs result then
             |       ^
          149|         result // {

       … while calling the 'isAttrs' builtin
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:148:10:
          147|     in
          148|       if isAttrs result then
             |          ^
          149|         result // {

       … from call site
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/lib/customisation.nix:138:16:
          137|     let
          138|       result = f origArgs;
             |                ^
          139|

       … while calling anonymous lambda
         at «github:cachix/devenv-nixpkgs/4267e705586473d3e5c8d50299e71503f16a6fb6»/pkgs/development/tools/delve/default.nix:1:1:
            1| { lib, buildGoModule, fetchFromGitHub, makeWrapper, stdenv }:
             | ^
            2|

       error: at «github:cachix/devenv/d3bb56ceaeef625712b050bd9343dbdb01bf37b1»/src/modules/languages/go.nix:9:21:
            8|   buildWithSpecificGo = pkg: pkg.override {
            9|     buildGoModule = pkgs."buildGo${goVersion}Module".override {
             |                     ^
           10|       go = cfg.package;

       error: attribute 'buildGo123Module' missing

       Did you mean one of buildGo121Module or buildGo122Module?
✔ Building shell in 0.9s.
```

**Version**
 
Not using flakes, version: `devenv 1.3.1 (aarch64-darwin)`

@sandydoo thank you for your quick response! 

However I am getting a new error:
```
• Building shell ...
• Using Cachix: devenv


copying path '/nix/store/4v964fw556w253glqjwbgshc382wylay-go-tools-2023.1.7-go-modules' from 'https://cache.nixos.org'
copying path '/nix/store/yd3dz3iwa8gsh5lbk43561aandj5dns3-gomodifytags-1.16.0-go-modules' from 'https://cache.nixos.org'

copying path '/nix/store/5axpg2g9ipfx9dab51fz3l9mdc5pj16a-gopls-0.15.2-go-modules' from 'https://cache.nixos.org'

copying path '/nix/store/z7j27l0l3ivzs7d2pxxw0lryjnvka6cq-gotests-1.6.0-go-modules' from 'https://cache.nixos.org'


copying path '/nix/store/w1z5ab4inaky4608g3fq0kr173w7mahr-impl-1.3.0-go-modules' from 'https://cache.nixos.org'


copying path '/nix/store/93j9hf9r9d08dc40i5nx2m2rr7qk3kf4-gotools-0.18.0-go-modules' from 'https://cache.nixos.org'
copying path '/nix/store/8xll7g7i1d8dagx2526csxsv5kkw1sjd-source' from 'https://cache.nixos.org'
copying path '/nix/store/6xjfd4hh4kd07bjnn8j318f7wlwkhhr3-source' from 'https://cache.nixos.org'

copying path '/nix/store/9bl7q9vl84rwkfphjis1xllvpffr3ja2-source' from 'https://cache.nixos.org'

copying path '/nix/store/hbzbfmpzpv95vf972s8rfc2yl1kbrx53-source' from 'https://cache.nixos.org'

copying path '/nix/store/hdcjancisq47hv88ls50v4zcjdf0qhgi-source' from 'https://cache.nixos.org'

copying path '/nix/store/lmmp6ll3rfqf2q0qmzrxp6annhqk8npb-source' from 'https://cache.nixos.org'

copying path '/nix/store/rlgfmw27w0jq6zivlpg9z02iw8swwfc0-source' from 'https://cache.nixos.org'
building '/nix/store/jjfv9lrzknxkm2lmksrbn0q1kscxzmvx-impl-1.3.0.drv'
building '/nix/store/0j1f2zjfz3jgaygivgaai38i8xcm6yy4-gomodifytags-1.16.0.drv'
building '/nix/store/h54wvj94zfz272q0y5s0dkj34ii3iz2s-gotests-1.6.0.drv'
Running phase: unpackPhase
unpacking source archive /nix/store/hdcjancisq47hv88ls50v4zcjdf0qhgi-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: buildPhase
Building subPackage .
Running phase: unpackPhase
unpacking source archive /nix/store/rlgfmw27w0jq6zivlpg9z02iw8swwfc0-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: buildPhase
building '/nix/store/ni9azarnyyn7mcq478rxxz84x5p7s6nd-go-tools-2023.1.7.drv'
Building subPackage .
Running phase: unpackPhase
unpacking source archive /nix/store/lmmp6ll3rfqf2q0qmzrxp6annhqk8npb-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: unpackPhase
unpacking source archive /nix/store/9bl7q9vl84rwkfphjis1xllvpffr3ja2-source
Running phase: buildPhase
building '/nix/store/rl4kwdz8x2h2k1q05rmh0ngfaikabq8q-gopls-0.15.2.drv'
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
building '/nix/store/b1q6shmwmzh7sdmlg5brvjbj939kzmw2-gotools-0.18.0.drv'
Running phase: buildPhase
Running phase: unpackPhase
unpacking source archive /nix/store/6xjfd4hh4kd07bjnn8j318f7wlwkhhr3-source
building '/nix/store/cdf8jcacb7yp2qv4l7fz1xpb5wdzqkid-delve-1.22.1.drv'
Running phase: unpackPhase
unpacking source archive /nix/store/8xll7g7i1d8dagx2526csxsv5kkw1sjd-source
Building subPackage .
Running phase: unpackPhase
unpacking source archive /nix/store/hbzbfmpzpv95vf972s8rfc2yl1kbrx53-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
source root is source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: buildPhase
Building subPackage ./.
Running phase: configurePhase
Running phase: buildPhase
Building subPackage ./cmd/dlv
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: buildPhase
Building subPackage ./analysis/code
Running phase: checkPhase
Running phase: checkPhase
Building subPackage ./benchmark/parse
Building subPackage ./gotests
Building subPackage ./gotests/process
ok  	github.com/fatih/gomodifytags	0.185s
Running phase: installPhase
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-gomodifytags-1.16.0.drv-0/ in /nix/store/b4admq13xm9za3rvgyfvjrsv0dblp0k4-gomodifytags-1.16.0...
patching script interpreter paths in /nix/store/b4admq13xm9za3rvgyfvjrsv0dblp0k4-gomodifytags-1.16.0
stripping (with command strip and flags -S) in  /nix/store/b4admq13xm9za3rvgyfvjrsv0dblp0k4-gomodifytags-1.16.0/bin
Building subPackage ./internal/goparser
Building subPackage ./internal/input
Building subPackage ./internal/models
Building subPackage ./internal/output
Building subPackage ./internal/render
Building subPackage ./internal/render/bindata
Building subPackage ./templates
Running phase: installPhase
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-gotests-1.6.0.drv-0/ in /nix/store/mbf3wmz5qqnjk00mwgcl8p6k91qmy4lm-gotests-1.6.0...
patching script interpreter paths in /nix/store/mbf3wmz5qqnjk00mwgcl8p6k91qmy4lm-gotests-1.6.0
stripping (with command strip and flags -S) in  /nix/store/mbf3wmz5qqnjk00mwgcl8p6k91qmy4lm-gotests-1.6.0/bin
# golang.org/x/telemetry/crashmonitor
vendor/golang.org/x/telemetry/crashmonitor/crash_go123.go:13:19: cannot use debug.SetCrashOutput (value of type func(f *os.File, opts debug.CrashOptions) error) as func(*os.File) error value in assignment
error: builder for '/nix/store/rl4kwdz8x2h2k1q05rmh0ngfaikabq8q-gopls-0.15.2.drv' failed with exit code 1;
       last 10 log lines:
       > Running phase: unpackPhase
       > unpacking source archive /nix/store/6xjfd4hh4kd07bjnn8j318f7wlwkhhr3-source
       > source root is source
       > Running phase: patchPhase
       > Running phase: updateAutotoolsGnuConfigScriptsPhase
       > Running phase: configurePhase
       > Running phase: buildPhase
       > Building subPackage ./.
       > # golang.org/x/telemetry/crashmonitor
       > vendor/golang.org/x/telemetry/crashmonitor/crash_go123.go:13:19: cannot use debug.SetCrashOutput (value of type func(f *os.File, opts debug.CrashOptions) error) as func(*os.File) error value in assignment
       For full logs, run 'nix-store -l /nix/store/rl4kwdz8x2h2k1q05rmh0ngfaikabq8q-gopls-0.15.2.drv'.
ok  	github.com/josharian/impl	0.602s
Running phase: installPhase
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-impl-1.3.0.drv-0/ in /nix/store/mfyiaasswfzyc5d83vqww6zhddlfnyz0-impl-1.3.0...
patching script interpreter paths in /nix/store/mfyiaasswfzyc5d83vqww6zhddlfnyz0-impl-1.3.0
stripping (with command strip and flags -S) in  /nix/store/mfyiaasswfzyc5d83vqww6zhddlfnyz0-impl-1.3.0/bin
Building subPackage ./analysis/edit
Building subPackage ./analysis/facts/deprecated
Building subPackage ./analysis/facts/directives
Running phase: installPhase
Building subPackage ./analysis/facts/generated
Building subPackage ./analysis/facts/nilness
Building subPackage ./blog
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-delve-1.22.1.drv-0/ in /nix/store/j0gw01h4yzca7mkzlpkrq8bdld6hy8bj-delve-1.22.1...
patching script interpreter paths in /nix/store/j0gw01h4yzca7mkzlpkrq8bdld6hy8bj-delve-1.22.1
stripping (with command strip and flags -S) in  /nix/store/j0gw01h4yzca7mkzlpkrq8bdld6hy8bj-delve-1.22.1/bin
Building subPackage ./analysis/facts/purity
Building subPackage ./analysis/facts/tokenfile
Building subPackage ./analysis/facts/typedness
Building subPackage ./analysis/lint
Building subPackage ./analysis/lint/testutil
Building subPackage ./analysis/report
Building subPackage ./cmd/keyify
Building subPackage ./cmd/staticcheck
Building subPackage ./blog/atom
Building subPackage ./cmd/auth/authtest
Building subPackage ./cmd/auth/cookieauth
Building subPackage ./cmd/auth/gitauth
Building subPackage ./cmd/auth/netrcauth
Building subPackage ./cmd/benchcmp
Building subPackage ./cmd/bisect
Building subPackage ./cmd/bundle
Building subPackage ./cmd/structlayout
Building subPackage ./cmd/structlayout-optimize
Building subPackage ./cmd/structlayout-pretty
Building subPackage ./cmd/unused
Building subPackage ./config
Building subPackage ./debug
Building subPackage ./go/ast/astutil
Building subPackage ./go/buildid
Building subPackage ./cmd/callgraph
Building subPackage ./go/gcsizes
Building subPackage ./go/ir
Building subPackage ./go/ir/irutil
Building subPackage ./go/loader
Building subPackage ./go/types/typeutil
Building subPackage ./internal/cmd/ast-to-pattern
Building subPackage ./internal/cmd/gogrep
Building subPackage ./cmd/compilebench
Building subPackage ./internal/cmd/irdump
Building subPackage ./cmd/deadcode
Building subPackage ./internal/diff/myers
Building subPackage ./internal/gosmith
# golang.org/x/telemetry/crashmonitor
vendor/golang.org/x/telemetry/crashmonitor/crash_go123.go:13:19: cannot use debug.SetCrashOutput (value of type func(f *os.File, opts debug.CrashOptions) error) as func(*os.File) error value in assignment
error: builder for '/nix/store/b1q6shmwmzh7sdmlg5brvjbj939kzmw2-gotools-0.18.0.drv' failed with exit code 1;
       last 10 log lines:
       > Building subPackage ./cmd/auth/gitauth
       > Building subPackage ./cmd/auth/netrcauth
       > Building subPackage ./cmd/benchcmp
       > Building subPackage ./cmd/bisect
       > Building subPackage ./cmd/bundle
       > Building subPackage ./cmd/callgraph
       > Building subPackage ./cmd/compilebench
       > Building subPackage ./cmd/deadcode
       > # golang.org/x/telemetry/crashmonitor
       > vendor/golang.org/x/telemetry/crashmonitor/crash_go123.go:13:19: cannot use debug.SetCrashOutput (value of type func(f *os.File, opts debug.CrashOptions) error) as func(*os.File) error value in assignment
       For full logs, run 'nix-store -l /nix/store/b1q6shmwmzh7sdmlg5brvjbj939kzmw2-gotools-0.18.0.drv'.
Building subPackage ./internal/passes/buildir
Building subPackage ./internal/renameio
Building subPackage ./internal/robustio
Building subPackage ./internal/sharedcheck
Building subPackage ./internal/sync
Building subPackage ./internal/testenv
Building subPackage ./knowledge
Building subPackage ./lintcmd
Building subPackage ./lintcmd/cache
Building subPackage ./lintcmd/runner
Building subPackage ./lintcmd/version
Building subPackage ./pattern
Building subPackage ./printf
Building subPackage ./quickfix
Building subPackage ./sarif
Building subPackage ./simple
Building subPackage ./staticcheck
Building subPackage ./staticcheck/fakejson
Building subPackage ./staticcheck/fakereflect
Building subPackage ./staticcheck/fakexml
Building subPackage ./structlayout
Building subPackage ./stylecheck
Building subPackage ./unused
Running phase: installPhase
Running phase: fixupPhase
checking for references to /private/tmp/nix-build-go-tools-2023.1.7.drv-0/ in /nix/store/14nvwbp1ymvc3qk4k4gs1b72njjz5a3q-go-tools-2023.1.7...
patching script interpreter paths in /nix/store/14nvwbp1ymvc3qk4k4gs1b72njjz5a3q-go-tools-2023.1.7
stripping (with command strip and flags -S) in  /nix/store/14nvwbp1ymvc3qk4k4gs1b72njjz5a3q-go-tools-2023.1.7/bin
error: 2 dependencies of derivation '/nix/store/hsa162cx3nm6ma2z58wvq066aaw6q4s6-devenv-profile.drv' failed to build
error:
       error: 3 dependencies of derivation '/nix/store/vw4b42aygjlxixrgpj3qz4jmwwrldrsr-devenv-shell-env.drv' failed to build
✔ Building shell in 24.1s.
Error:   × Failed to run command `/nix/store/89cx3xdqqm65q2pdq984bipr0bbgmv4s-nix-2.24-devenv/bin/nix --show-trace --extra-experimental-features nix-command --extra-experimental-features flakes --option warn-dirty false --keep-going --max-jobs 8 --option eval-cache false
  │ --option extra-substituters https://devenv.cachix.org --option extra-trusted-public-keys devenv.cachix.org-1:w1cLUi8dv3hnoSPGAuibQv+f9TZLr6cv/Hm9XgU50cw= nixpkgs-python.cachix.org-1:hxjI7pFxTyuTHn2NkvWCrAUcNZLNS3ZAvfYNuYifcEU= print-dev-env --profile /Users/
  │ fgruntjes/projects/teda-tech/crossplane-management-cluster/.devenv/gc/shell -vv --log-format internal-json`
  ╰─▶ Nix command failed: exit status: 1
 ```

--- a/src/modules/languages/go.nix
+++ b/src/modules/languages/go.nix
@@ -3,13 +3,9 @@
 let
   cfg = config.languages.go;
 
-  goVersion = (lib.versions.major cfg.package.version) + (lib.versions.minor cfg.package.version);
-
-  buildWithSpecificGo = pkg: pkg.override {
-    buildGoModule = pkgs."buildGo${goVersion}Module".override {
-      go = cfg.package;
-    };
-  };
+  # Override the buildGoModule function to use the specified Go package.
+  buildGoModule = pkgs.buildGoModule.override { go = cfg.package; };
+  buildWithSpecificGo = pkg: pkg.override { inherit buildGoModule; };
 in
 {
   options.languages.go = {
@@ -45,7 +41,7 @@ in
       (buildWithSpecificGo pkgs.gotests)
     ];
 
-    hardeningDisable = (lib.optional (cfg.enableHardeningWorkaround) "fortify");
+    hardeningDisable = lib.optional (cfg.enableHardeningWorkaround) "fortify";
 
     env.GOROOT = cfg.package + "/share/go/";
     env.GOPATH = config.env.DEVENV_STATE + "/go";

